relay_messages.go

```diff
@@ -249,6 +250,10 @@ func (f *lazyCallReq) Arg2Iterator() (arg2.KeyValIterator, error) {
 	return arg2.NewKeyValIterator(f.Payload[f.arg2StartOffset:f.arg2EndOffset])
 }
 
+func (f *lazyCallReq) Arg2Append(keyVals []relay.KeyVal) {
+	f.arg2Appends = keyVals
```

Comment:
all we've done is shift the allocation from inside of the append method to the caller. i'm not sure if that actually helps. One advantage is that we can now pre-allocate a slice that's sized exactly for the case of > 1 allocation but that's not the common case today

Can you try the array trick? Use a backing array in the `lazyCallReq` and create a slice that's backed by the array and append to that?

This interface is also a little less clear -- since it now replaces any previously added appends, so calling it twice means the last write wins.